src/incubator/WheelPicker/index.tsx

```diff
@@ -39,47 +46,104 @@ export interface WheelPickerProps {
   /**
    * Event, on active row change
    */
-  onChange: (index: number, item?: ItemProps) => void;
+  onValueChange?: (item: string | number, index: number) => void;
+  /**
+   * Container's ViewStyle, height is computed according to itemHeight * numberOfVisibleRows
+   */
+  style?: Omit<ViewStyle, 'height'>;
+  /**
+   * Support passing items as children props
+   */
+  children?: JSX.Element | JSX.Element[];
+  /**
+   * WheelPicker initial value, can be ItemProps.value, number as index
+   */
+  selectedValue: ItemProps | number | string;
 }
 
-const WheelPicker = ({items, itemHeight = 48, activeTextColor, inactiveTextColor, textStyle, onChange: onChangeEvent}: WheelPickerProps) => {
-  const height = itemHeight * 5;
+const WheelPicker = ({
+  items: propItems,
+  itemHeight = 44,
+  numberOfVisibleRows = 5,
+  activeTextColor,
+  inactiveTextColor,
+  textStyle,
+  onValueChange,
+  style,
+  children,
+  selectedValue
+}: WheelPickerProps) => {
   const scrollView = useRef<Animated.ScrollView>();
   const [offset] = useValues([0], []);
+  let scrollOffset = useRef<number>(0).current;
   const onScroll = onScrollEvent({y: offset});
-  
-  const listSize = items?.length || 0;
-  const middleIndex = useMiddleIndex({itemHeight, listSize});
+
+  const {height, items, shouldControlComponent, index: currentIndex, getRowItemAtOffset} = usePresenter({
+    selectedValue,
+    items: propItems,
+    children,
+    itemHeight,
+    preferredNumVisibleRows: numberOfVisibleRows
+  });
+
+  useEffect(() => {
+    controlComponent();
+  });
+
+  /**
+   * The picker is a controlled component. This means we expect the
+   * to relay on `selectedValue` prop to be our
+   * source of truth - not the picker current value.
+   * This way, you can control disallow or mutate selection of some values.
+   */
+  const controlComponent = () => {
```

Comment:
Verify that initial value works.. right now it always takes the first one.

Reply:
So it works, it based on the value given as `items` or `children`, I fixed the demo so it be more clear it's not an index..
`_.map(months, value => ({label: value, value: value}));`